codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -551,7 +586,17 @@ protected void onConnectionError(ChannelHandlerContext ctx, Throwable cause, Htt
         if (http2Ex == null) {
             http2Ex = new Http2Exception(INTERNAL_ERROR, cause.getMessage(), cause);
         }
-        goAway(ctx, http2Ex).addListener(new ClosingChannelFutureListener(ctx, ctx.newPromise()));
+
+        ChannelPromise promise = ctx.newPromise();
+        ChannelFuture future = goAway(ctx, http2Ex);
+        switch (http2Ex.shutdownHint()) {
+        case SOFT_SHUTDOWN:
```

Comment:
nit: Since "graceful` doesn't really refer to the HTTP/2 definition here, maybe we should also call this "graceful" for now ... since it basically means the same thing as the other uses of "graceful".  WDYT?